cdap-app-fabric/src/test/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeServiceTest.java

```diff
@@ -72,18 +78,19 @@ public ProgramLiveInfo getLiveInfo(Id.Program programId) {
       }
 
       @Override
-      protected Location getArtifactLocation(ProgramDescriptor descriptor) throws IOException,
-                                                                                  ArtifactNotFoundException {
-        // Just return some dummy location. It is not used by the test.
-        return Locations.toLocation(TEMP_FOLDER.newFile());
-      }
-
-      @Override
       protected Program createProgram(CConfiguration cConf, ProgramRunner programRunner,
                                       ProgramDescriptor programDescriptor,
                                       Location programJarLocation, File tempDir) throws IOException {
         return program;
       }
+
+      @Override
+      protected ArtifactDetail getArtifactDetail(ArtifactId artifactId) throws IOException, ArtifactNotFoundException {
+        co.cask.cdap.api.artifact.ArtifactId id = new co.cask.cdap.api.artifact.ArtifactId(
+          "dummy", new ArtifactVersion("1.0"), ArtifactScope.USER);
+        return new ArtifactDetail(new ArtifactDescriptor(id, null),
```

Comment:
For the `ArtifactDetail` class, it is not. But in the context of this test, it doesn't matter as it is never used.
Updated the code to provide a dummy Location instead of `null` to safeguard against future changes.